docs: refactor install docs

ZeRego opened the pull request:
- write a new doc page with all the install instructions
- redirect all the old docs to new page
- update old page references to new page
- set sidebar structure so we can have the old pages hidden
- remove required env vars default values from the docker compose and set the same default values in the install script

TuringLovesDeathMetal commented on .env:
```diff
@@ -3,12 +3,9 @@ PORT=8080
 PGHOST=db
 PGPORT=5432
 PGUSER=postgres
-PGPASSWORD=password
```
@ZeRego - did you mean to change this file 🤔 not sure if this is related to the install docs

ZeRego replied:
I'm removing the default values for important fields from the docker compose. 
This will remove the risk of people deploying to production with default values and be easily hacked. 
This way we force people to setup their own database password and Lightdash secret.

These default values will still be set when using the install script for a quick install.

cc: @owlas

owlas replied:
One user actually requested that we move this to `.env.template` that way it won't be used by default

ZeRego replied:
I think the rest of the defaults are fine and we now have clear docs showing what vars they can set and what are their default values.

TuringLovesDeathMetal commented on README.md:
```diff
@@ -66,10 +66,7 @@ cd lightdash
-* [Setup Lightdash manually](https://docs.lightdash.com/get-started/setup-a-local-dbt-project) (*requires your own dbt project*)
-* [Setup Lightdash in Heroku with dbt cloud](https://docs.lightdash.com/get-started/setup-a-dbt-cloud-project) (*requires your own dbt project on dbtCloud*)
-* [Setup Lightdash in Heroku with GitHub](https://docs.lightdash.com/get-started/setup-a-github-project) (*requires your own dbt project on GitHub*)
+* [Install Lightdash](https://docs.lightdash.com/get-started/setup-lightdash/install-lightdash) (*requires your own dbt project*)
```
🤩

TuringLovesDeathMetal commented on docker-compose.yml:
```diff
@@ -17,7 +17,6 @@ services:
       - LIGHTDASH_INSTALL_ID=${LIGHTDASH_INSTALL_ID}
       - LIGHTDASH_INSTALL_TYPE=${LIGHTDASH_INSTALL_TYPE:-docker_image}
     volumes:
-      - "${DBT_PROFILES_DIR}:/usr/app/profiles"
```
same with this - is this meant to be in this PR?

ZeRego replied:
We no longer use the profiles.yml from their local machine so this volume isn't necessary anymore.

owlas replied:
So new default is to use the UI and not to use a local dbt project?

ZeRego replied:
> So new default is to use the UI and not to use a local dbt project?
@owlas The default and only option to setup the warehouse connection is via the UI, using a local dbt project or not.

owlas replied:
Ah yes! This is just the profiles dir - TY

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+
+:::caution Production deployments
+
+For production deployments check the [Deploy manually](#deploy-manually) instructions.
```
```suggestion
If you want to be able to share your Lightdash project with your team, you'll need to create a production deployment. 

For instructions on how to set up a production deployment check the [Deploy manually](#deploy-manually) docs. (After all, data is more fun with friends 🧑‍🤝‍🧑)
```

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+
+## Deploy locally with our installation script
+
+This is the recommended way to install and run Lightdash in your local machine.
```
Again, please update if this is wrong!
```suggestion
If you only want to run Lightdash in your local machine (that means, you won't be able to share the things you build in Lightdash with anyone else), then this is how we'd recommend you install it. 
```

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+
+:::info
+
+This script is going to install docker and docker compose. If you want do use a docker alternative, see [Deploy manually](#deploy-manually) instructions.
```
```suggestion
This script is going to install Docker and Docker Compose. If you want do use a Docker alternative, see [Deploy manually](#deploy-manually) instructions.
```

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+
+:::info
+
+If you're using a local dbt project, during the cli, select the option **"with local dbt"**
```
```suggestion
If you're using a local dbt project, when you're asked `how you want to setup Lightdash`, select the option **"with local dbt"**
```

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+
+## Deploy to Heroku
+
+Click to deploy the latest version of Lightdash to heroku. This provides a secure, production-ready deployment.
```
What's the difference between a production-ready deployment and a production deployment? Like, if I wanted to share my Lightdash project with my team, could I use the one-click-deploy to Heroku? Or what would I have to do to get it production-ready?

ZeRego replied:
> What's the difference between a production-ready deployment and a production deployment?

There isn't any difference. I can remove the `-ready` part.

> Like, if I wanted to share my Lightdash project with my team, could I use the one-click-deploy to Heroku? 

yes

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+import TOCInline from '@theme/TOCInline';
+
+# Install Lightdash
+
```
Would be great to add a little summary/table of contents type thing here! Suggested text (but not sure what to write for the one-click deploy bit). 

Also, I _think_ this is what the difference between local vs. production deployment is, but please correct me if I'm wrong! What I'm trying to get across here is like "here's what that means you'd be limited to doing"

```suggestion
There are three ways you can install Lightdash:
1. If you want to run Lightdash in your local machine (that means, you won't be able to share the things you build in Lightdash with anyone else), then you can [get started with our installation script](#deploy-locally-with-our-installation-script).
2. If you want to have Lightdash.... # I don't know what to write here because I'm not sure why I'd choose the 1-click deploy option!
3. If you want to be able to share your Lightdash project with your team, then you can set up a production deployment of Lightdash by following our instructions on [how to deploy Lightdash manually](#deploy-manually).
```

ZeRego replied:
@TuringLovesDeathMetal 
Why people would use Heroku deployments ?
- they don't wan't to run Lightdash in their machine so running in Heroku is the simplest way to try Lightdash 
- their company uses Heroku to manage their infrastructure instead of AWS/GCP/etc, so they would run their Lightdash there

ZeRego replied:
> What I'm trying to get across here is like "here's what that means you'd be limited to doing"
This might be a bit misleading because the limitations don't depend just on our instructions but where they are using them.
The instructions for the manual deployment can be used in any place that supports docker, being AWS/GCP or their local machine.
E.g  users can deploy Lightdash manually in their local machine, so they won't be able to share what they built.

We can only recommend what instructions to follow when developing and for production but in the end it will depend on their expertise, setup and infrastructure.

TuringLovesDeathMetal replied:
Okay, that's really helpful information. 

Is it accurate to say:
1. If you want to deploy Lightdash in production, you can use the instructions for the one-click-deploy with Heroku _**or**_ the manually deploy 
2. one-click deploy to Heroku is the simplest way to deploy a Lightdash project in production
3. if you want to deploy a production Lightdash instance on a cloud platform that supports docker and _**isn't**_ Heroku (e.g. AWS or GCP), then you can use our manual deploy instructions 
4. if you want to deploy Lightdash on your local machine, then you should use our install script. But, you can also use our manual deploy instructions if you're really keen.

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+
+![screenshot](../assets/heroku-success.png)
+
+#### 3. Setup your admin account
```
```suggestion
#### 3. Setup your admin account and Lightdash project
```

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+
+#### 3. Setup your admin account
+
+As the first user for Lightdash, you'll be prompted to create an admin account to secure your instance.
```
```suggestion
As the first user for Lightdash, you'll be prompted to create an admin account to secure your instance. Then, you'll get started with creating your Lightdash project. 

You can follow along with the instructions for setting up your Lightdash project here: #Add page to instructions for Lightdash project setup guide
```

TuringLovesDeathMetal reviewed (changes requested):
really really awesome job writing this all up and coordinating all of the changes across a bajillion places, @ZeRego 🚀 

I've added some suggestions to sort of "highlight" the differences between the different install types. 

I've also added a little bit about the like "continuing your installation process" but not sure if we should add it until after the next set of docs exists...

visch commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
```diff
@@ -0,0 +1,173 @@
+# Lightdash port: 8080
+
+docker run --name lightdash-app \
+    -e PGHOST=192.168.0.1 -e PGPORT=5432 \
```
I wonder what most setups would have here for an IP address? It all depends on the setup :/ one negative of not using docker compose

owlas commented on docs/sidebars.js:
```diff
@@ -10,17 +10,45 @@
-  // But you can create a sidebar manually
-  /*
-  tutorialSidebar: [
+  lightdash: [
```
smart, nice change!

owlas replied:
<img width="370" alt="Screenshot 2021-10-05 at 14 22 32" src="https://user-images.githubusercontent.com/11660098/136031429-c4d122c4-4a19-421e-821a-335bf5022938.png">
 is the menu supposed to have double nesting? 
 
 Getting started > Setting up Lightdash > Install Lightdash

ZeRego replied:
Yes, you can see the entire menu in the ticket description #502
Right now is weird because we don't have the other pages yet. Going to work on the `Connect a project` next.

owlas commented on install.sh:
```diff
@@ -390,17 +390,17 @@ start_docker
 # script doesn't exit because this command looks like it failed to do it's thing.
 if [[ $setup_type == 'local_dbt' ]]; then
-    sudo LIGHTDASH_INSTALL_ID="$INSTALLATION_ID" LIGHTDASH_INSTALL_TYPE="$LIGHTDASH_INSTALL_TYPE" PORT="$port" DBT_PROJECT_DIR="$dbt_project_dir" docker-compose -f docker-compose.yml up --detach --remove-orphans || true
+    sudo LIGHTDASH_INSTALL_ID="$INSTALLATION_ID" LIGHTDASH_INSTALL_TYPE="$LIGHTDASH_INSTALL_TYPE" PORT="$port" PGPASSWORD="password" LIGHTDASH_SECRET="not very secret" DBT_PROJECT_DIR="$dbt_project_dir" docker-compose -f docker-compose.yml up --detach --remove-orphans || true
```
Why inline this? it's much harder to find out what's going on when you're setting up

ZeRego replied:
I also found it to not be ideal but it was the only way that worked. 
I might have been doing something wrong so I'm open to suggestions.

owlas reviewed (commented):
Looks strong! Couple of questions

TuringLovesDeathMetal commented on docs/docs/get-started/setup-lightdash/install-lightdash.mdx:
````diff
@@ -0,0 +1,175 @@
+    -e PGDATABASE=postgres -e LIGHTDASH_SECRET="not very secret" \
+    -p 8080:8080 \
+    -d lightdash/lightdash:latest
+```
````
```suggestion
```

----

## Getting set up on Lightdash Cloud

Want to have a production-ready Lightdash instance, but not comfortable with self-hosting? You can [sign up to our waitlist for Lightdash Cloud.](https://hamzah905194.typeform.com/to/HFlicx4i)

With Lightdash Cloud, you get a Lightdash instance that's production-ready (so you can share all your insights with your team) but you don't have to worry about any of the setup for hosting the instance yourself. So, if you're not too sure about all of this container, Heroku and cloud-platforms business, or you're just not interested in the overhead of hosting your BI yourself, then [Lightdash Cloud](https://hamzah905194.typeform.com/to/HFlicx4i) might be a good option for you :)
```

ZeRego commented:
> I've also added a little bit about the like "continuing your installation process" (but only in the one-click deploy to Heroku bit). Maybe we shouldn't add this until after the next set of docs exists?
> 
> ^I've added this as a comment in this issue: https://github.com/lightdash/lightdash/projects/1#card-69931181

@TuringLovesDeathMetal I'm going to leave the message you proposed even missing the link for the connect project page. I'm going to work on that doc next and fix the link there :)

TuringLovesDeathMetal reviewed (approved):
Looks gucci!
